### DISCUSSION TITLE
Atom dependency from a different store

### DISCUSSION THREAD

**Discussion by fjcalzado** [pmndrs/jotai#2058] [Q&A] [answered]
Recently I realized that, while `useAtom` hook exposes an option to determine the scope (store) for the value we are retrieving:
```
useAtomValue(myGlobalAtom, { store: getDefaultStore() });
```

we cannot do the same when resolving a derived atom dependency:
```
const myDerivedAtom = atom<number>(get => {
  const globalValue = get(myGlobalAtom); // <= There is no way to indicate the getter to retrieve this atom value from a specific store
  ...
});
```
Perhaps I am wrong but I guess `globalValue` dependency is being retrieved from the closest provider/store (the same where `myDerivedAtom` is being mounted). Is this correct? Is this a design/architecture decision or a limitation? Cause I find it very limiting when using multiple nested providers.

**Comment by dai-shi** [comment, marked as answer]
It's a design choice. Atom abstraction is independent from store and reusable across multiple stores.

Jotai core doesn't handle anything with multiple stores. What you may find interesting is jotai-molecules: https://jotai.org/docs/integrations/molecules

**Reply by dai-shi** [reply]
> Jotai core doesn't handle anything with multiple stores.

I mean it supports multiple stores. Just that stores are totally isolated.

**Comment by fjcalzado** [comment]
Thanks for your answer @dai-shi !

> It's a design choice

That's what I thought.

> Jotai core doesn't handle anything with multiple stores.

But Store and Provider appears under Core section in documentation. Thus, I thought you could drop multiple Providers with its own Stores ... but then I found what I explained in the OP and made me regret of doing so. 

I feel like sooner or later we need to 'isolate' atom values in certain places of the application for the solely purpose of "clearing all atoms by remounting" as it is stated in the Provider documentation. It is also [discussed here](https://github.com/pmndrs/jotai/discussions/682) (I am sorry I found this thread late).

I will take a look at `molecule` concept, but I have the feeling that molecules will add a lot of boilerplate, especially when having dozens or hundreds of atoms, cause you need to `useMolecule` first in order to access the scoped atoms.

**Reply by shulcsm** [reply]
If you only want to read and your store is global you can read directly from it. But other than that i'm not sure how cross-store updates would work.

**Reply by fjcalzado** [reply]
Yes, you can read from global/default store directly like this for example:
```
useAtomValue(myGlobalAtom, { store: getDefaultStore() });
```
but you couldn't make that global value a dependency for a derived atom under a Provider, as the value will be fetched from that provider store:
```
const myDerivedAtom = atom(get => get(myGlobalAtom));
...

const ShowDerived = () => {
  const derivedValue = useAtomValue(myDerivedAtom);
  return <p>{derivedValue}</p>;
}

...

<Provider>
  <ShowDerived />
</Provider>

```

**Comment by dai-shi** [comment]
I just developed another util for this kind of use cases. Hope someone can try.
https://github.com/jotaijs/jotai-scope/pull/2
